src/id.rs

```diff
@@ -77,32 +109,61 @@ impl FromStr for Id {
     }
 }
 
-impl From<u32> for Id {
-    fn from(n: u32) -> Self {
+impl From<Inner> for Id {
+    fn from(n: Inner) -> Self {
         Self { n }
     }
 }
 
 #[cfg(test)]
 mod tests {
+    use std::i64;
+
     use super::*;
 
     #[test]
-    fn convert_u32_to_id_and_back() {
+    fn convert_inner_to_id_and_back() {
         let id = Id::from(0);
-        assert_eq!(id.to_string(), "aaaaaa");
-        assert_eq!(id.as_u32(), 0);
+        assert_eq!(id.to_string(), "aaaaaaaaaaa");
+        assert_eq!(id.as_inner(), 0);
+        assert_eq!(Id::from_str(&id.to_string()).unwrap(), id);
+
+        let id = Id::from(-1);
+        assert_eq!(id.to_string(), "++++++++++p");
+        assert_eq!(id.as_inner(), -1);
+        assert_eq!(Id::from_str(&id.to_string()).unwrap(), id);
 
         let id = Id::from(0xffffffff);
-        assert_eq!(id.to_string(), "+++++d");
-        assert_eq!(id.as_u32(), 0xffffffff);
+        assert_eq!(id.to_string(), "aaaaap++++p");
+        assert_eq!(id.as_inner(), 0xffffffff);
+        assert_eq!(Id::from_str(&id.to_string()).unwrap(), id);
+
+        let id = Id::from(i64::MAX);
+        assert_eq!(id.to_string(), "F+++++++++p");
+        assert_eq!(id.as_inner(), i64::MAX);
+        assert_eq!(Id::from_str(&id.to_string()).unwrap(), id);
+
+        let id = Id::from(i64::MIN);
+        assert_eq!(id.to_string(), "Gaaaaaaaaaa");
+        assert_eq!(id.as_inner(), i64::MIN);
+        assert_eq!(Id::from_str(&id.to_string()).unwrap(), id);
     }
 
     #[test]
     fn convert_id_from_string() {
-        assert!(Id::from_str("abDE+-").is_ok());
+        /* Support ID generated in the old 32-bit format */
+        //assert!(Id::from_str("abDE+-").is_ok());
```

Comment:
Why is this outcommented? Also, please use proper Rust comments, i.e. `//` instead of `/* */`.